go/tools/gazelle/packages/package.go

```diff
@@ -136,3 +137,171 @@ func (ts *PlatformStrings) firstGoFile() string {
 	}
 	return ""
 }
+
+// addFile adds the file described by "info" to a target in the package "p" if
+// the file is buildable.
+//
+// "cgo" tells whether a ".go" file in the package contains cgo code. This
+// affects whether C files are added to targets.
+// "buildTags" is a set of tags that are true on all platforms.
+// "platforms" is a map from platform names (labels referencing config_settings)
+// to sets of tags that are true on those platforms. The tag sets may include
+// "buildTags".
+//
+// An error is returned if a file is buildable but invalid (for example, a
+// test .go file containing cgo code). Files that are not buildable will not
+// be added to any target (for example, .txt files).
+func (p *Package) addFile(info fileInfo, cgo bool, buildTags map[string]bool, platforms PlatformConstraints) error {
+	var t *Target
+	if info.isXTest {
+		if info.isCgo {
+			return fmt.Errorf("%s: use of cgo in test not supported", info.path)
+		}
+		t = &p.XTest
+	} else if info.isTest {
+		if info.isCgo {
+			return fmt.Errorf("%s: use of cgo in test not supported", info.path)
+		}
+		t = &p.Test
+	} else if info.isCgo || cgo && (info.category == cExt || info.category == hExt || info.category == csExt) {
+		t = &p.CgoLibrary
+	} else if info.category == goExt || info.category == sExt || !cgo && info.category == hExt {
```

Comment:
Yes, it is redundant. Thought it would make the condition more readable, but maybe it's just confusing.